internal/expr/function.go

```diff
@@ -1,658 +1,95 @@
 package expr
```

Comment:
The functions logic is becoming pretty big and deserves its own package I believe. How about moving all of it to `internal/expr/functions`?